library/network/src/modules/SuSEFirewall.rb

```diff
@@ -1617,6 +1617,71 @@ def SetIgnoreLoggingBroadcast(zone = nil, bcast)
       nil
     end
 
+    # Function returns list of allowed ports for zone and protocol
+    #
+    # @param [String] zone
+    # @param [String] protocol
+    # @return	[Array<String>] of allowed ports
+    def GetAllowedServicesForZoneProto(zone, protocol)
+
+      Yast.import "SuSEFirewallServices"
+
+      ports = []
+      protocol.downcase!
+
+      @SETTINGS[zone][:ports].each do |p|
+        port_proto = p.split("/")
+        ports << port_proto[0] if port_proto[1] == protocol
+      end
+      @SETTINGS[zone][:protocols].each do |p|
+        ports << p if protocol == "ip"
```

Comment:
This got me quite confused. I think it would be better to 
- rename the variable *ports* to *result* (because it is so polymorphic)
- `result = @SETTINGS[zone][:protocols] if protocol == "ip"`